alert-common/src/main/java/com/synopsys/integration/alert/common/descriptor/config/field/validators/ValidationResult.java

```diff
@@ -31,45 +31,76 @@
 
 public class ValidationResult {
 
-    private Collection<String> errors;
+    private final Collection<String> errors;
+    private final Collection<String> warnings;
 
     public static ValidationResult success() {
         return new ValidationResult();
     }
 
     public static ValidationResult errors(Collection<String> errors) {
-        return new ValidationResult(errors);
+        return new ValidationResult(errors, List.of());
     }
 
     public static ValidationResult errors(String... errors) {
-        return new ValidationResult(Arrays.asList(errors));
+        return new ValidationResult(Arrays.asList(errors), List.of());
+    }
+
+    public static ValidationResult warnings(Collection<String> warnings) {
+        return new ValidationResult(List.of(), warnings);
+    }
+
+    public static ValidationResult warnings(String... warnings) {
+        return new ValidationResult(List.of(), Arrays.asList(warnings));
     }
 
     public static ValidationResult of(ValidationResult... validationResults) {
         Collection<String> validationErrors = Arrays.stream(validationResults)
                                                   .map(ValidationResult::getErrors)
                                                   .flatMap(Collection::stream)
                                                   .collect(Collectors.toList());
-        return new ValidationResult(validationErrors);
+        Collection<String> validationWarnings = Arrays.stream(validationResults)
+                                                    .map(ValidationResult::getWarnings)
+                                                    .flatMap(Collection::stream)
+                                                    .collect(Collectors.toList());
+        return new ValidationResult(validationErrors, validationWarnings);
```

Comment:
You can abstract this further by creating a private method that is passed a function. The function will either do getErrors or getWarnings. You'd then call the method twice here with different functions passed.